holmes/plugins/toolsets/investigator/core_investigation.py

```diff
@@ -0,0 +1,159 @@
+            required=True,
+        ),
+    }
+
+    # Print a nice table to console/log
+    def print_tasks_table(self, tasks):
+        if not tasks:
+            logging.info("No tasks in the investigation plan.")
+            return
+
+        status_icons = {
+            "pending": "[ ]",
+            "in_progress": "[~]",
+            "completed": "[✓]",
+        }
+
+        max_id_width = max(len(str(task.id)) for task in tasks)
+        max_content_width = max(len(task.content) for task in tasks)
+        max_status_display_width = max(
+            len(f"{status_icons[task.status.value]} {task.status.value}")
+            for task in tasks
+        )
+
+        id_width = max(max_id_width, len("ID"))
+        content_width = max(max_content_width, len("Content"))
+        status_width = max(max_status_display_width, len("Status"))
+
+        # Build table
+        separator = f"+{'-' * (id_width + 2)}+{'-' * (content_width + 2)}+{'-' * (status_width + 2)}+"
+        header = f"| {'ID':<{id_width}} | {'Content':<{content_width}} | {'Status':<{status_width}} |"
+
+        # Log the table
+        logging.info("Updated Investigation Tasks:")
+        logging.info(separator)
+        logging.info(header)
+        logging.info(separator)
+
+        for task in tasks:
+            status_display = f"{status_icons[task.status.value]} {task.status.value}"
+            row = f"| {task.id:<{id_width}} | {task.content:<{content_width}} | {status_display:<{status_width}} |"
+            logging.info(row)
+
+        logging.info(separator)
+
+    def _invoke(self, params: Dict) -> StructuredToolResult:
+        try:
+            todos_data = params.get("todos", [])
+
+            tasks = []
+
+            for todo_item in todos_data:
+                if isinstance(todo_item, dict):
+                    task = Task(
+                        id=todo_item.get("id", str(uuid4())),
+                        content=todo_item.get("content", ""),
+                        status=TaskStatus(todo_item.get("status", "pending")),
+                    )
+                    tasks.append(task)
+
+            logging.info(f"Tasks: {len(tasks)}")
+
+            # Store tasks in session storage
+            todo_manager = get_todo_manager()
+            session_id = params.get("investigation_id", "")
+            todo_manager.update_session_tasks(session_id, tasks)
+
+            self.print_tasks_table(tasks)
+
+            formatted_tasks = todo_manager.format_tasks_for_prompt(session_id)
+
+            response_data = f"✅ Investigation plan updated with {len(tasks)} tasks. Tasks are now stored in session and will appear in subsequent prompts.\n\n"
+            if formatted_tasks:
+                response_data += formatted_tasks
+            else:
+                response_data += "No tasks currently in the investigation plan."
+
+            return StructuredToolResult(
+                status=ToolResultStatus.SUCCESS,
+                data=response_data,
```

Comment:
Do we think sending the task list to the AI as part of the tool call + prompt isn't redundant?

Reply:
might be, but this will be used by the ui for presentation
we might change it afterwards